game/view/sector.lua

```diff
@@ -211,6 +214,37 @@ function SectorView:getBodySprite(body)
   return body_sprite
 end
 
+
+function SectorView:setBodySprite(body, draw)
+  self.body_sprites[body:getId()] = draw
+end
+
+function SectorView:getBodyDialogue(body, i, j, pi, pj)
+  local id = body:getId()
+
+  --Get appropriate position for dialogue box
+  local side
+  if pj <= j then
+    side = "right"
+  else
+    side = "left"
+  end
+
+  local dialogue_box = self.body_dialogues[id]
+  if not dialogue_box then
+    dialogue_box = DIALOGUEBOX(body, i - 1, j - 1, side)
+    self.body_dialogues[id] = dialogue_box
+  else
+    dialogue_box:setSide(side)
+  end
+  return dialogue_box
+end
+
+function SectorView:resetBodyDialogue(body)
+  local id = body:getId()
+  self.body_dialogues[id] = nil
+end
+
```

Comment:
This is the kind of stuff that smells like it should be somewhere else, and not here. I know SectorView is a handful, but bloating it even more is contributing to the problem. Avoid writing more functionality into it.

Reply:
As discussed in the meeting, we are leaving dialogue draw on sector, and will try to reduce it in another PR.